src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -158,13 +151,18 @@ public static int getTransportCost(final Collection<Unit> units) {
     return mapping;
   }
 
-  private static List<Unit> sortByTransportCapacityAscending(final Collection<Unit> transports) {
+  private static List<Unit> sortByTransportCapacityAscendingThenMovesDescending(final Collection<Unit> transports) {
     final Comparator<Unit> transportCapacityComparator = new Comparator<Unit>() {
       @Override
       public int compare(final Unit o1, final Unit o2) {
         final int capacityLeft1 = TransportTracker.getAvailableCapacity(o1);
         final int capacityLeft2 = TransportTracker.getAvailableCapacity(o2);
-        return capacityLeft1 - capacityLeft2;
+        if (capacityLeft1 != capacityLeft2) {
+          return Integer.compare(capacityLeft1, capacityLeft2);
```

Comment:
Is this swapping o1 and o2?
When:  capacityLeft2 > capacityLeft1,
we'll get a negative number.

Unless I'm mistaken, which I might be, but `Integer.compare(capacityLeft1, capacityLeft2);` would return +1 when capacityLeft2 is larger?

Also of note, the `1` and `2` are then later swapped for the movement:
`return Integer.compare(movementLeft2, movementLeft1);`

If the ordering is correct, a comment in here to help the reader out to know why we swapped the ordering would be of benefit.